src/ir_interpreter/interpret_ir.ml

```diff
@@ -316,6 +316,11 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
         check_call_conv_arg env exp v2 call_conv;
         last_region := exp.at; (* in case the following throws *)
         f (context env) v2 k
+      | MutReadPrim id, [] ->
```

Comment:
It's odd that this operation doesn't actually do the read, it just returns the binding of the location, exactly like the VarE case.

I guess the reading is actually done elsewhere.

To me this suggest it might be best to just add a mutability field to VarE itself, instead of a misleadingly named MutReadPrim that also needs to be special cased everywhere because it has an identifier payload that contributes, eg. to freevar calc.

Reply:
Hmm, I wonder how many places I have touch in the IR-related source files. Having `VarE (Const, id)` is a bit clumsy and wild-carding as `VarE _` (or `VarE (_, id)`) still carries the risk of messing up with mutable code.
I am not opposed to this at all, though. And I appreciate its charm. Let's see how it works out.

Reply:
Turns out, I didn't touch _that_ many places, see 1273d3ea8.